core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -44,12 +41,11 @@ public FilteredQueryParser() {
     }
 
     @Override
-    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+    public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
-        Query query = Queries.newMatchAllQuery();
-        Query filter = null;
-        boolean filterFound = false;
+        QueryBuilder query = FilteredQueryBuilder.generateDefaultQuery();
```

Comment:
maybe we could pass in null to the builder since the default is already handled there, that way the odd generateDefaultQuery could become private too :)